inc/classes/subscriber/Tools/class-cloudflare-subscriber.php

```diff
@@ -0,0 +1,888 @@
+						);
+
+						return new \WP_Error( 'cloudflare_invalid_auth', $msg );
+					}
+				}
+				$msg = __( 'Incorrect Cloudflare Zone ID.', 'rocket' );
+
+				$msg .= ' ' . sprintf(
+					/* translators: %1$s = opening link; %2$s = closing link */
+					__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+					// translators: Documentation exists in EN, FR; use localized URL if applicable.
+					'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
+					'</a>'
+				);
+				return new \WP_Error( 'cloudflare_invalid_auth', $msg );
+			}
+
+			if ( true === $cf_zone->success ) {
+				$zone_found = false;
+				$site_url   = get_site_url();
+
+				if ( function_exists( 'domain_mapping_siteurl' ) ) {
+					$site_url = domain_mapping_siteurl( $site_url );
+				}
+
+				if ( ! empty( $cf_zone->result ) ) {
+					$parsed_url = wp_parse_url( $site_url );
+					if ( false !== strpos( strtolower( $parsed_url['host'] ), $cf_zone->result->name ) ) {
+						$zone_found = true;
+					}
+				}
+
+				if ( ! $zone_found ) {
+					$msg = __( 'It looks like your domain is not set up on Cloudflare.', 'rocket' );
+
+					$msg .= ' ' . sprintf(
+						/* translators: %1$s = opening link; %2$s = closing link */
+						__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+						// translators: Documentation exists in EN, FR; use localized URL if applicable.
+						'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
+						'</a>'
+					);
+					return new \WP_Error( 'cloudflare_wrong_zone_id', $msg );
+				}
+
+				return true;
+			}
+		} catch ( \Exception $e ) {
+			$msg  = __( 'Incorrect Cloudflare email address or API key.', 'rocket' );
+			$msg .= ' ' . sprintf(
+				/* translators: %1$s = opening link; %2$s = closing link */
+				__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+				// translators: Documentation exists in EN, FR; use localized URL if applicable.
+				'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
+				'</a>'
+			);
+			return new \WP_Error( 'cloudflare_invalid_auth', $msg );
+		}
+	}
+
+	/**
+	 * Get a Cloudflare\Api instance & the zone_id corresponding to the domain
+	 *
+	 * @since 2.8.21 Get the zone ID from the options
+	 * @since 2.8.18 Add try/catch to prevent fatal error Uncaugh Exception
+	 * @since 2.8.16 Update to Cloudflare API v4
+	 * @since 2.5
+	 *
+	 * @return Object Cloudflare instance & zone_id if credentials are correct, WP_Error otherwise
+	 */
+	public static function get_instance() {
```

Comment:
`get_instance()` is generally used to get the class instance, so it's a bit confusing here. What do you think of renaming to:

```suggestion
	public static function get_cloudflare_instance() {
```